File: src/main/java/org/model/Competences/Competence.java
package org.model.Competences;

public class Competence {

    private int id;
    private String category;
    private String type;
    private String description;


    private String niveau;
    private int anneesExperience;
    private String certification;
    private String statut;


    private String email;

    public Competence() {}


    public Competence(int id, String category, String type, String description,
                      String niveau, int anneesExperience, String certification, String statut,
                      String email) {
        this.id = id;
        this.category = category;
        this.type = type;
        this.description = description;
        this.niveau = niveau;
        this.anneesExperience = anneesExperience;
        this.certification = certification;
        this.statut = statut;
        this.email = email;
    }


    public Competence(int id, String category, String type, String description) {
        this(id, category, type, description, "Débutant", 0, null, "En cours", null);
    }

    public int getId() { return id; }
    public String getCategory() { return category; }
    public String getType() { return type; }
    public String getDescription() { return description; }

    public String getNiveau() { return niveau; }
    public int getAnneesExperience() { return anneesExperience; }
    public String getCertification() { return certification; }
    public String getStatut() { return statut; }
    public String getEmail() { return email; }


    public void setId(int id) { this.id = id; }
    public void setCategory(String category) { this.category = category; }
    public void setType(String type) { this.type = type; }
    public void setDescription(String description) { this.description = description; }

    public void setNiveau(String niveau) { this.niveau = niveau; }
    public void setAnneesExperience(int anneesExperience) { this.anneesExperience = anneesExperience; }
    public void setCertification(String certification) { this.certification = certification; }
    public void setStatut(String statut) { this.statut = statut; }
    public void setEmail(String email) { this.email = email; }
}

File: src_backup_phpstan/Entity/competence/Competence.php
<?php

namespace App\Entity\competence;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: "competences")]
class Competence
{
    // ============================================================
    // 🔑 ID (clé primaire)
    // ============================================================
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private ?int $id = null;

    // ============================================================
    // 📌 CATEGORY
    // ============================================================
    #[ORM\Column(type: "string", length: 100)]
    private ?string $category = null;

    // ============================================================
    // 📌 TYPE
    // ============================================================
    #[ORM\Column(type: "string", length: 100)]
    private ?string $type = null;

    // ============================================================
    // 📌 DESCRIPTION
    // ============================================================
    #[ORM\Column(type: "text", nullable: true)]
    private ?string $description = null;

    // ============================================================
    // 📌 NIVEAU
    // ============================================================
    #[ORM\Column(type: "string", length: 50)]
    private ?string $niveau = null;

    // ============================================================
    // 📌 ANNÉES D’EXPÉRIENCE
    // ============================================================
    #[ORM\Column(name: "annees_experience", type: "integer")]
    private ?int $anneesExperience = null;

    // ============================================================
    // 📌 CERTIFICATION
    // ============================================================
    #[ORM\Column(type: "string", length: 255, nullable: true)]
    private ?string $certification = null;

    // ============================================================
    // 📌 STATUT
    // ============================================================
    #[ORM\Column(type: "string", length: 50)]
    private ?string $statut = null;

    // ============================================================
    // 📌 EMAIL
    // ============================================================
    #[ORM\Column(type: "string", length: 150)]
    private ?string $email = null;

    // ============================================================
    // 🧠 GETTERS & SETTERS
    // ============================================================

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCategory(): ?string
    {
        return $this->category;
    }

    public function setCategory(string $category): self
    {
        $this->category = $category;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getNiveau(): ?string
    {
        return $this->niveau;
    }

    public function setNiveau(string $niveau): self
    {
        $this->niveau = $niveau;
        return $this;
    }

    public function getAnneesExperience(): ?int
    {
        return $this->anneesExperience;
    }

    public function setAnneesExperience(int $anneesExperience): self
    {
        $this->anneesExperience = $anneesExperience;
        return $this;
    }

    public function getCertification(): ?string
    {
        return $this->certification;
    }

    public function setCertification(?string $certification): self
    {
        $this->certification = $certification;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;
        return $this;
    }
}